From: "Matthew Wilcox (Oracle)" <willy@infradead.org>
Subject: mm: make is_free_buddy_page() take a const argument
Date: Tue, 26 Mar 2024 17:10:27 +0000

This function does not modify its argument; let the callers know that so
they can make better optimisation decisions.

Link: https://lkml.kernel.org/r/20240326171045.410737-6-willy@infradead.org
Signed-off-by: Matthew Wilcox (Oracle) <willy@infradead.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 include/linux/page-flags.h |    2 +-
 mm/page_alloc.c            |    8 ++++----
 2 files changed, 5 insertions(+), 5 deletions(-)

--- a/include/linux/page-flags.h~mm-make-is_free_buddy_page-take-a-const-argument
+++ a/include/linux/page-flags.h
@@ -1088,7 +1088,7 @@ static inline bool is_page_hwpoison(cons
 	return folio_test_hugetlb(folio) && PageHWPoison(&folio->page);
 }
 
-extern bool is_free_buddy_page(struct page *page);
+bool is_free_buddy_page(const struct page *page);
 
 PAGEFLAG(Isolated, isolated, PF_ANY);
 
--- a/mm/page_alloc.c~mm-make-is_free_buddy_page-take-a-const-argument
+++ a/mm/page_alloc.c
@@ -6751,16 +6751,16 @@ void __offline_isolated_pages(unsigned l
 /*
  * This function returns a stable result only if called under zone lock.
  */
-bool is_free_buddy_page(struct page *page)
+bool is_free_buddy_page(const struct page *page)
 {
 	unsigned long pfn = page_to_pfn(page);
 	unsigned int order;
 
 	for (order = 0; order < NR_PAGE_ORDERS; order++) {
-		struct page *page_head = page - (pfn & ((1 << order) - 1));
+		const struct page *head = page - (pfn & ((1 << order) - 1));
 
-		if (PageBuddy(page_head) &&
-		    buddy_order_unsafe(page_head) >= order)
+		if (PageBuddy(head) &&
+		    buddy_order_unsafe(head) >= order)
 			break;
 	}
 
_
